pkg/controller/machineautoscaler/machineautoscaler_controller.go

```diff
@@ -43,11 +44,45 @@ var SupportedTargetGVKs = []schema.GroupVersionKind{
 	{Group: "cluster.k8s.io", Version: "v1alpha1", Kind: "MachineDeployment"},
 }
 
+// Config represents the configuration for a reconciler instance.
+type Config struct {
+	// The namespace for MachineAutosclaers and their targets.
+	Namespace string
+}
+
+// ObjectReference represents a CrossVersionObjectReference
+// identifying a target resource with additional helper methods.
+type ObjectReference v1alpha1.CrossVersionObjectReference
```

Comment:
Coudn't we use https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/objectreference.go?

Reply:
Yep. Missed that. Done.